config-tool/pkg/lib/shared/storage_validators.go

```diff
@@ -143,6 +144,58 @@ func ValidateStorage(opts Options, storageName string, storageType string, args
 			errors = append(errors, err)
 		}
 
+	case "STSS3Storage":
+
+		// Check bucket name
+		if ok, err := ValidateRequiredString(args.S3Bucket, "DISTRIBUTED_STORAGE_CONFIG."+storageName+".s3_bucket", fgName); !ok {
+			errors = append(errors, err)
+		}
+
+		roleArn := args.STSRoleArn
+		sess := session.Must(session.NewSession(&aws.Config{
+			Credentials: awscredentials.NewStaticCredentials(args.STSUserAccessKey, args.STSUserSecretKey, ""),
+		}))
+		svc := sts.New(sess)
+		roleToAssumeArn := roleArn
+		durationSeconds := int64(3600)
+		assumeRoleInput := &sts.AssumeRoleInput{
+			RoleArn:         aws.String(roleToAssumeArn),
+			RoleSessionName: aws.String("quay"),
```

Comment:
is this documented anywhere as the session name?

Reply:
This session name can be anything

Reply:
yes, understood, and it should be fairly obvious from the name of the session, but some customers under more regulatory requirements may want to see documentation of the STS token session retrieval